photon-client/src/main/scala/com/linkedin/photon/ml/util/IOUtils.scala

```diff
@@ -365,4 +366,44 @@ object IOUtils {
 
     builder.mkString
   }
+
+  /**
+   * Save GAME per-group evaluation in terms of (random effect id, evaluation value) to HDFS.
+   *
+   * @param sc The Spark context
+   * @param outputDir The directory in HDFS where to save the evaluation
+   * @param evaluationResultsOpt An Option of evaluation results
+   * @param logger The logger instance for the application
+   */
+  def saveGameEvaluationToHDFS(
+    sc: SparkContext,
```

Comment:
Since you are using Spark, and the number of groups can be large, potentially in millions, why don't you use spark.dataframe to save in avro format. Faster and more scalable.

Reply:
Output format is changed to Avro.